06_gpu_and_ml/yolo/yolo.py

```diff
@@ -0,0 +1,288 @@
+# ---
+# runtimes: ["runc", "gvisor"]
+# ---
+#
+# # Training and Inferencing YOLOv10
+#
+# Example by [@AnirudhRahul](https://github.com/AnirudhRahul)
+#
+# [YOLOv10](https://docs.ultralytics.com/models/yolov10/) is the most modern version of the popular "You Only Look Once" ("YOLO") computer vision model, 
+# which has been trained to perform bounding box object detection.
+#
+# In this example, we use the V10 model via the [Ultralytics package](https://docs.ultralytics.com/). We will:
+# - Fine tune on a custom dataset, in our case the DoclayNet dataset for object detection within documents.
+# - Deploy an inference endpoint.
+
+# For commercial use, sure to consult the [ultralytics software license](https://docs.ultralytics.com/#yolo-licenses-how-is-ultralytics-yolo-licensed).
+
+# ## Building Our Image
+#
+# The example uses the Ultralytics package to run training inference and the Huggingface dataset package to load DoclayNet data.
+# 
+# All packages are installed into a Pytorch 2.3.1 base image
```

Comment:
Any reason we can't use `debian_slim`? I imagine it's because ultralytics wants a `CUDA_HOME`, whereas PyTorch installation via `pip` does a `site-packages` install.

If so, we should at least prefer the NVIDIA Docker container that we recommend in the guide and install PyTorch on top.